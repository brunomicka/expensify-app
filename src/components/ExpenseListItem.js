import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({ description, amount, createdAt, note, id }) => {
    return (
        <div>
            <p><strong>Description:</strong> <Link to={`/edit/${id}`}>{description}</Link></p>
            <p>
                <strong>Amount:</strong> 
                {numeral(amount/100).format('$ 0,0.00')}
            </p>
            <p>
                <strong>Created At:</strong> 
                {moment(createdAt).format('MMMM Do, YYYY')}
            </p>
            <p><strong>Note:</strong> {note}</p>
            <hr />
        </div>
    )
};

export default ExpenseListItem;