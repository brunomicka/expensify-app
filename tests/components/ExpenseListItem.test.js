import React from 'react';
import ExpenseListItem  from '../../src/components/ExpenseListItem';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem with an expense', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[1]} key={expenses[1].id}/>);
    expect(wrapper).toMatchSnapshot();
})