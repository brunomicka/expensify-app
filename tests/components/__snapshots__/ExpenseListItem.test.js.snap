// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with an expense 1`] = `
<div>
  <p>
    <strong>
      Description:
    </strong>
     
    <Link
      replace={false}
      to="/edit/2"
    >
      Rent
    </Link>
  </p>
  <p>
    <strong>
      Amount:
    </strong>
    $ 1,095.00
  </p>
  <p>
    <strong>
      Created At:
    </strong>
    December 27th, 1969
  </p>
  <p>
    <strong>
      Note:
    </strong>
     
  </p>
  <hr />
</div>
`;
